rest/base.ts

```diff
@@ -57,8 +57,8 @@ export class Base {
   public static config: ConfigInterface;
 
   public static apiVersion: string;
-  protected static resourceNames: string[] = [];
-  protected static pluralNames: string[] = [];
+  protected static resourceNames: {[key: string]: string}[] = [];
```

Comment:
We should probably create an interface for this so we can name the `singular` and `plural` fields.